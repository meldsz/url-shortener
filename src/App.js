import React, { useState } from 'react';
import { TextField, Button } from '@material-ui/core'
import './App.css';

function App() {
  const [longUrl, setLongUrl] = useState('');
  const [shortUrl, setShortUrl] = useState('');

  return (
    <div className="App">
      <div className="app-title">
        URL Shortener
      </div>
      <div>
        <div className="input-container">
          <TextField variant="outlined"
            className="input-url"
            fullWidth
            label="Enter your URL here"
            onChange={(e) => setLongUrl(e.target.value)} />
        </div>
        <div><Button color="primary"
          variant="contained"
          onClick={submitUrl} >Submit</Button></div>
      </div>

      {shortUrl &&
        <div className="display-short-url">
          Shortened URL: <span><a href={longUrl} target="_blank">{shortUrl}</a></span>
        </div>
      }

    </div>
  );
}

export default App;
