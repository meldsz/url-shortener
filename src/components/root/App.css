.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 50vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(20px + 2vmin);
  color: white;
}

.link-icon {
  font-size: 50px;
}

.input-container {
  margin:3% 10%;
}

.display-short-url  {
  padding: 15px;
}

.form-container {
  text-align: center;
  height: 15vh;
}

.btn-container {
  margin-left: 45%;
}