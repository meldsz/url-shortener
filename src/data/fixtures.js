// export const url = {
//     'shortUrl': 'http://localhost/jasgh',
//     'originalUrl': 'http://url.com',
//     'shortId': 'jasgh'
// }

export const url = {
    "_id": "5d8b84d112592214d81b6304",
    "originalUrl": "http://google.com",
    "shortUrl": "http://localhost:5000/lyMkFDYh",
    "shortId": "lyMkFDYh",
    "mtime": "2019-09-25T15:16:33.710Z",
    "ctime": "2019-09-25T15:16:33.814Z",
    "__v": 0
}