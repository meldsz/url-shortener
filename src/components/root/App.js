import React, { Component } from 'react';
import { FormGroup, Form, FormControl, Button, } from 'react-bootstrap'
import './App.css';
import { endpoints, host } from '../../utils/endpoints';
import ShortUrl from '../short-url/ShortUrl';

class App extends Component {
  constructor() {
    super();
    this.state = {
      longUrl: '',
      shortUrl: ''
    }
  }

  submitUrl = async () => {
    const postData = { originalUrl: this.state.longUrl, host };
    try {
      let response = await fetch(endpoints.longUrl, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(postData)
      });
      response = await response.json();
      response &&
        this.setState({ shortUrl: response.shortUrl });
      // this.redirectToShortUrl(response.shortId);
    } catch (err) {
      console.log('Error:', err);
    }

  }

  redirectToShortUrl = async (shortId) => {
    let response = await fetch(endpoints.getShortUrl(shortId));
    await response.json();

  }

  render() {
    const { longUrl, shortUrl } = this.state;
    return (
      <div className="container App">
        <div className="col App-header">
          <div className="row">
            <span className="glyphicon glyphicon-link link-icon"></span>
          </div>

          <div className="row">
            URL Shortener
          </div>
        </div>


        <div className="col form-container">
          <Form>
            <FormGroup>
              <div className="row input-container">
                <FormControl placeholder="Enter your URL here"
                  onChange={(e) => this.setState({
                    longUrl: e.target.value
                  })} />
              </div>
              <div className="row btn-container">
                <Button variant="primary"
                  className="submit-btn"
                  onClick={this.submitUrl}>Submit</Button>
              </div>
            </FormGroup>
          </Form>

        </div>


        {
          shortUrl &&
          <div className="row display-short-url">
            <ShortUrl longUrl={longUrl} shortUrl={shortUrl} />
          </div>
        }


      </div>
    );
  }


}

export default App;
