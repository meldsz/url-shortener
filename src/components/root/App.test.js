import React from 'react';
import { mount } from 'enzyme';
import App from './App';
import sinon, { fakeServer } from 'sinon';
import { url } from '../../data/fixtures';

const inputUrl = 'http://dadaaaaa.com';
const apiUrl = 'http://localhost:5000/api/urls';
const shortUrl = "http://localhost:5000/lyMkFDYh";

describe('app', () => {
  let server;
  beforeEach(() => {
    server = fakeServer.create();
    server.respondWith('POST',
      apiUrl,
      [
        200,
        { 'Content-Type': 'application/json' },
        JSON.stringify(url)
      ]
    );

  });


  describe('app component', () => {
    const app = mount(<App />);
    const mockEvent = { target: { value: inputUrl } };

    it('app renders correctly', () => {
      expect(app).toMatchSnapshot();
    });

    it('initial state', () => {
      expect(app.state()).toEqual({ longUrl: '', shortUrl: '' });
    });

    it('does not render ShortUrl component initially', () => {
      expect(app.find('ShortUrl').exists()).toBe(false);
    });

    it('should have input box', () => {
      expect(app.find('FormControl')).toHaveLength(1);
    });

    it('renders a Button `Submit` ', () => {
      expect(app.find('.submit-btn').at(0).props().children).toEqual('Submit');
    });

    describe('updating the input box longUrl', () => {

      beforeEach(() => {
        app.find('FormControl').simulate('change', mockEvent);
      })

      it('update longUrl in the state', () => {
        expect(app.state().longUrl).toEqual(inputUrl);
      });

      describe('submitting the form with longUrl', () => {

        beforeEach(done => {
          global.fetch = jest.fn().mockImplementation(() => {
            return Promise.resolve(new Response(JSON.stringify(url)));
          });
          app.find('Button').simulate('click');
          setTimeout(done)
        });

        it('updates the state with shortUrl obtained from api', () => {
          expect(global.fetch).toHaveBeenCalledTimes(1);
          expect(app.state().shortUrl).toEqual(shortUrl);
          global.fetch.mockClear();
        })
      });

    });


  });



});
