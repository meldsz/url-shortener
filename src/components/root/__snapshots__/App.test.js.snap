// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app app component app renders correctly 1`] = `ReactWrapper {}`;
